util/object_pool.c

```diff
@@ -903,11 +903,12 @@ static int objpool_borrow_int(comdb2_objpool_t op, void **objp, long nanosecs,
 
     /* update access history */
     rec = (pooled_object *)hash_find(op->history, objp);
+    Pthread_mutex_unlock(&op->data_mutex);
+
     rec->active = 1;
     ++rec->nborrows;
     rec->tid = pthread_self();
```

Comment:
I think that these three need to be done under the lock , otherwise another thread can access this record and find this info inconsistent.
However there is room for improvement to do the creation of rec outside of lock object_create(), and only add it under lock.

Reply:
It's okay. `get_impl` makes sure that nothing can grab the same object again after it returns (because it's done while holding the lock), which implies the caller's exclusive access to the object's history.